java/arcs/sdk/Reference.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+@file:Suppress("FunctionName") // Allow pseudo-constructors.
+
+package arcs.sdk
+
+import arcs.core.data.RawEntity
+import arcs.sdk.common.ReferenceId
+import arcs.sdk.storage.StorageKey
+
+/** Represents a reference (ie. pointer) to an [Entity]. */
+typealias Reference = arcs.core.storage.Reference
+
+/**
+ * Creates a new [Reference] to an [Entity] with the given [entityId] within the region identified
+ * by [regionKey].
+ */
+fun Reference(entityId: ReferenceId, regionKey: StorageKey) =
```

Comment:
I thought we would derive regionKey / storageKeys in the allocator? If so, is there any way we can avoid putting this in the public API?

Reply:
That's fair, I could drop this pseudo-constructor and instead either have one that accepts the entity to be referenced, or rely on codegen to add a `.toReference()` method on Entity implementations,  as well as a `Reference.toMyEntity()` extension function.